pkg/writetarget/write_target_monitor.go

```diff
@@ -30,7 +30,7 @@ const (
 // includes decoding messages as specific types and deriving metrics based on the decoded messages.
 // TODO: Report decoding uses the same ABI for EVM and Aptos, however, future chains may need a different
 // decoding scheme. Generalize this in the future to support different chains and decoding schemes.
-func NewMonitor(ctx context.Context, lggr logger.Logger) (*monitor.BeholderClient, error) {
+func NewMonitor(ctx context.Context, lggr logger.Logger, decodeFn func(m *wt.WriteConfirmed) ([]*registry.FeedUpdated, error)) (*monitor.BeholderClient, error) {
```

Comment:
Pre-existing, but why does this take a context?

Reply:
fixed